src/components/Controls.svelte

```diff
@@ -1,31 +1,38 @@
 <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined:opsz,wght,FILL,GRAD@48,700,0,0" />
 
 <script lang="ts">
-	import { createEventDispatcher } from 'svelte';
-	import { construct_svelte_component } from 'svelte/internal';
 	import ControlStore from '../stores/ControlStore';
-	
+	import { onDestroy } from 'svelte';
+
 	// context vs store;
 	// can only play one song at a time; player controls was extracted
-	const dispatch = createEventDispatcher();	
 	export let songid: string; // required to make modifications to songid (ie add to playlist and more)
-	export let audio: HTMLAudioElement;       //is audio track for song, is an HTML Audio Element
+	export let audio: HTMLAudioElement; //is audio track for song, is an HTML Audio Element
 	export let added: boolean; //whether or not song is added to playlist
 	export let liked: boolean; //whether or not song is favorited
 	export let position: number;
-	let status = {
-		PLAY: 'PLAY',
-		PAUSE: 'PAUSE'
+	export let savedPosition : number;
+
+	let updatePosition = setInterval(() => {
+		position = Math.max(savedPosition, Math.round(audio.currentTime));
+		console.log(position)
+		if(position >= audio.duration){
+			currStatus = status.PAUSE;
+			position = 0;
+		}
+	}, 500);
+
+	enum status{
+		PLAY,
```

Comment:
[video on why enums are bad](https://www.youtube.com/watch?v=jjMbPt_H3RQ)
[video on types vs interfaces][(https://www.youtube.com/watch?v=zM9UPcIyyhQ)

don't tale the videos as the whole truth. They're just generating discussions, make valid points, but must consider everything. imo, some cases, it makes sense to use `enum`, but you just gotta use it correctly. (ie instead of just saying `PLAY`, you gotta do `PLAY = "play"` (you should almost always assign your enum's))